fs/_ftp_parse.py

```diff
@@ -81,17 +84,21 @@ def parse_line(line):
     return None
 
 
-def _parse_time(t, formats):
-    t = " ".join(token.strip() for token in t.lower().split(" "))
-
-    _t = None
+def _find_suitable_format(t, formats):
     for frmt in formats:
         try:
-            _t = time.strptime(t, frmt)
+            if time.strptime(t, frmt):
+                return frmt
         except ValueError:
             continue
-    if not _t:
+    return None
+
+
+def _parse_time(t, formats):
+    frmt = _find_suitable_format(t, formats)
+    if frmt is None:
         return None
+    _t = time.strptime(t, frmt)
```

Comment:
Hmmm, seems a bit wasteful to do `time.strptime` (at least) twice? Once in `_find_suitable_format` and again in `_parse_time`.

Reply:
That statement is correct though in my opinion, "a bit" is negligible. A call to `strptime` costs about 4 microseconds on my machine. I find `_find_suitable_format` easier to read than something like `_decode_time_in_suitable_format` and I'd argue that readability should always be the higher priority over resource efficiency in Python - otherwise one should be using C anyway.

That being said, if you feel like that has to be changed to be accepted, I'll change it.

Reply:
I'll leave it up to the other maintainers...

Reply:
I'd be in favour of a single function as well. I don't think it hurts readability that much, and the functions are quite highly coupled anyway.

Reply:
Done